charmcraft.yaml

```diff
@@ -46,3 +46,12 @@ actions:
     - model
     - prompt
     additionalProperties: false
+  pull:
+    description: "Downloads and serves the model you specify. Find available models here: https://ollama.com/library"
+    params:
+      model:
+        type: string
+        description: "The name of the model to pull"
+    required:
+    - model
```

Comment:
I would make `model` optional, and e.g. use lamma3.1 or one that was already pulled, if lamma3.1 wasn't.

Reply:
Done here: https://github.com/Thinking-Dragon/ollama-juju-charm/pull/7